app/src/main/java/com/codesoom/assignment/controllers/ArgumentNotValidAdvice.java

```diff
@@ -0,0 +1,46 @@
+package com.codesoom.assignment.controllers;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.codesoom.assignment.dto.ErrorResponse;
+import com.codesoom.assignment.dto.UserData;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * MethodArgumentNotValidException 예외를 처리한다.
+ */
+@ControllerAdvice
+public class ArgumentNotValidAdvice {
+    /**
+     * MethodArgumentNotValidException을 처리한다.
+     *
+     * @return 어디서(url경로, Http 메서드) 어떤 에러가 발생하였으며, 어떻게 해결할 수 있을지를 리턴한다.
+     *
+     * @see UserController#create(UserData)
+     *
+     */
```

Comment:
이 정도로 충분하겠죠. `MethodArgumentNotValidException` 을 javadoc에 적는 것은 불필요하고, 커플링도 발생합니다. 예외 이름이 바뀌거나 다른 예외를 이 목적으로 쓰게 되었을 때 javadoc을 수정하지 않으면 잘못된 내용이 남게 되겠죠.

```suggestion
    /**
     * 던져진 "잘못된 메소드 입력값 예외"를 받아 에러 응답을 리턴합니다.
     *
     * @return 예외에 대한 내용이 담긴 응답
     */
```

한편 `@see UserController#create(UserData)`는 있어서는 안 되는 내용입니다. 이 예외를 던지는 곳이 어디어디인지 이 클래스에 알려져서는 안 됩니다. 얘는 공이 떠오르는 것을 기다리는 야구선수처럼 하늘만 보고 있다가 `MethodArgumentNotValidException`가 던져 올려지면 받는 역할만 하는 것입니다.

한편 예외 이름이 너무 포괄적이군요. 좀 더 구체적인 이름을 가진 예외 클래스를 사용하거나 적절히 번역해야 할 필요가 있어 보입니다.

Reply:
넵 수정하겠습니다.
예외 이름이 너무 포괄적이라는 말씀은 `MethodArgumentNotValidException`의 이름을 의미하는것인가요?
커스텀 예외를 만들어 사용하라는 의미로 이해하였는데, 제가 이해한것이 맞는지 궁금합니다.

Reply:
맞아요. 김형욱님의 PR에 제가 남긴 코멘트도 함께 읽어보시면 도움이 되겠네요.
https://github.com/CodeSoom/spring-week5-assignment-1/pull/27#discussion_r705302968